### PR TITLE
define a service

### PR DESCRIPTION
**Pull request by sbawaska**
defining a service and loosening the content-type check to enable functions publishing to streams.

### CODE REVIEW

File: pkg/client/client.go [hunk 1]
```diff
@@ -68,7 +69,7 @@ func NewStreamClient(gateway string, topic string, acceptableContentType string)
 
 func (lc *StreamClient) Publish(ctx context.Context, payload io.Reader, key io.Reader, contentType string, headers map[string]string) (PublishResult, error) {
 	m := serialization.Message{}
-	if lc.acceptableContentType != contentType { // TODO support smarter compatibility (eg subtypes)
+	if !strings.HasPrefix(contentType, lc.acceptableContentType) { // TODO support smarter compatibility (eg subtypes)
```
Comment (ericbottard) [inline pkg/client/client.go]: This is not the proper way to support subtypes.
I'd rather leave it as is for now, unless this is blocking you (it shouldn't)

File: config/riff-http-gateway.yaml [hunk 2]
```diff
@@ -29,6 +29,25 @@ subjects:
   name: default
   namespace: riff-system
 ---
+apiVersion: v1
```
Comment (ericbottard) [inline config/riff-http-gateway.yaml:32]: This should be put in a separate file. Sometimes we want a ClusterIP service, sometimes a LoadBalancer, etc...

File: pkg/client/client.go (hunk 1 above)
Comment (sbawaska) [reply pkg/client/client.go]: I am using RestTemplate to post request from function. RestTemplate uses `application/json;utf-8` by default. I didn't want to change it there as it made the function look complicated.

Comment (ericbottard) [reply pkg/client/client.go]: At the very least, chop until `;` on both sides and continue to do an equality check

File: config/riff-http-gateway.yaml (hunk 2 above)
Comment (sbawaska) [reply config/riff-http-gateway.yaml:32]: We could look into parameterizing the service type during installation. (may-be from helm). I think for now, ClusterIP should work.

File: pkg/client/client.go (hunk 1 above)
Comment (sbawaska) [reply pkg/client/client.go]: done!

**Pull request merged**